src/OpenTelemetry.AutoInstrumentation/Loading/AspNetInitializer.cs

```diff
@@ -0,0 +1,56 @@
+// <copyright file="AspNetInitializer.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#if NETFRAMEWORK
+
+using System;
+using System.Threading;
+using OpenTelemetry.AutoInstrumentation.Plugins;
+
+namespace OpenTelemetry.AutoInstrumentation.Loading;
+
+internal class AspNetInitializer
+{
+    private readonly PluginManager _pluginManager;
+
+    private int _initialized;
+
+    public AspNetInitializer(LazyInstrumentationLoader lazyInstrumentationLoader, PluginManager pluginManager)
+    {
+        _pluginManager = pluginManager;
+        lazyInstrumentationLoader.Add(new AspNetMvcInitializer(InitializeOnFirstCall));
+        lazyInstrumentationLoader.Add(new AspNetWebApiInitializer(InitializeOnFirstCall));
+    }
+
+    private void InitializeOnFirstCall(ILifespanManager lifespanManager)
+    {
+        if (Interlocked.Exchange(ref _initialized, value: 1) != default)
+        {
+            // InitializeOnFirstCall() was already called before
+            return;
+        }
+
+        var instrumentationType = Type.GetType("OpenTelemetry.Instrumentation.AspNet.AspNetInstrumentation, OpenTelemetry.Instrumentation.AspNet");
```

Comment:
Because of the dependencies issue, especially with S.D.DS, it seems that is possible at the moment to get null here. The code should check for that and log as appropriate.

Reply:
That happens so late in the runtime, that we are probably crashed way earlier if that happens.

Reply:
Even if it throws then it should be handled here https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/abeb491aabac80b62588fdb87f7536f7520e1b71/src/OpenTelemetry.AutoInstrumentation/Loading/LazyInstrumentationLoader.cs#L69, right?